feat(treemap): Extend tree map API. (#1089)
uber/react-vis: docs/treemap.md

Scale properties for the `color` scale. If `color` property is not passed in the data object, each new section of the chart gets the next color (e. g. the `'category'` scale is applied).
Please refer to [Scales and Data](scales-and-data.md) for more information about scales.

> #### getSize

Reviewer: Maybe include you example in the docs via a <!-- INJECT:"SimpleTreemapWithLink" --> type thing

Type: `function`

- Should accept arguments (leaf)

Pass in a function that will be used for get size.

#### getColor (optional)

Type: `function`

- Should accept arguments (leaf)

> Pass in a function that will be used for get color.

Reviewer 2: ```suggestion
Pass in a function that will be used to get color.
```

#### getLabel (optional)

Type: `function`

- Should accept arguments (leaf)

> Pass in a function that will be used for get label.

Reviewer 2: ```suggestion
Pass in a function that will be used to get label.
```

#### getContent (optional)

Type: `function`

- Should accept arguments (leaf)

> Pass in a function that will be used for get inner content of leaf.

Reviewer 2: ```suggestion
Pass in a function that will be used to get inner content of leaf.
```

Can return a string or react component (JSX)

#### getChildren (optional)

Type: `function`

- Should accept arguments (leaf)

Pass in a function that will be used for get children

> #### titileHeight (optional)

Reviewer 2: ```suggestion
#### titleHeight (optional)
```

Type: `number`

- Height of title in leafs in pixels

> Determines top padding for leafs names.

Reviewer 2: ```suggestion
Determines top padding for leaf names.
```

#### paddingInner (optional)

Type: `number`

- Size of gap in pixels

> Determines gap between lefs. If not setted gap will be same as padding value.

Reviewer 2: ```suggestion
Determines gap between leaves. If not set gap will be same as padding value.
```

> Set 0 for remove inner gap.

Reviewer 2: ```suggestion
Set 0 to remove inner gap.
```
